overlord/snapstate/handlers.go

```diff
@@ -1969,6 +1969,10 @@ func (m *SnapManager) doDiscardSnap(t *state.Task, _ *tomb.Tomb) error {
 			t.Errorf("cannot discard snap namespace %q, will retry in 3 mins: %s", snapsup.InstanceName(), err)
 			return &state.Retry{After: 3 * time.Minute}
 		}
+		err = m.backend.RemoveSnapInhibitLock(snapsup.InstanceName())
```

Comment:
This doesn't match the description of the PR (linking/unlinking the snap affects the lock), I supposed the description needs updating. Nonetheless, can you explain why this is not done in unlink-snap?

Reply:
It's done here to ensure you don't end up "inheriting" a lock after snap remove / snap install. If you somehow manage to remove a snap the lock, even if present, should be removed as well.

As for unlink, when the snap is unlinked, it's not necessarily being removed. Putting this code there under a special case felt wrong. For analogy we have discard-snap-ns as a distinct task, and not as special code in unlink for the same reason.